Failed: UnfoldResourceSourceSpec

https://jenkins.akka.io:8498/job/akka-multi-node-nightly/5991/testReport/junit/akka.stream.scaladsl/UnfoldResourceSourceSpec/Unfold_Resource_Source_must_fail_when_close_throws_exception/

https://jenkins.akka.io:8498/job/akka-nightly/5506/testReport/junit/akka.stream.scaladsl/UnfoldResourceSourceSpec/Unfold_Resource_Source_must_fail_when_close_throws_exception/

https://jenkins.akka.io:8498/job/akka-multi-node/2767/testReport/junit/akka.stream.scaladsl/UnfoldResourceSourceSpec/Unfold_Resource_Source_must_fail_when_close_throws_exception/

https://jenkins.akka.io:8498/job/akka-nightly/5495/testReport/junit/akka.stream.scaladsl/UnfoldResourceSourceSpec/Unfold_Resource_Source_must_fail_when_close_throws_exception/

```
Error Message

java.lang.AssertionError: timeout (3 seconds) waiting for 1 messages on ErrorFilter(class akka.stream.testkit.Utils$TE,None,Left(),false)
Stacktrace

sbt.ForkMain$ForkError: java.lang.AssertionError: timeout (3 seconds) waiting for 1 messages on ErrorFilter(class akka.stream.testkit.Utils$TE,None,Left(),false)
	at akka.testkit.EventFilter.intercept(TestEventListener.scala:114)
	at akka.stream.scaladsl.UnfoldResourceSourceSpec$$anonfun$1$$anonfun$apply$mcV$sp$9$$anonfun$apply$9.apply(UnfoldResourceSourceSpec.scala:194)
	at akka.stream.scaladsl.UnfoldResourceSourceSpec$$anonfun$1$$anonfun$apply$mcV$sp$9$$anonfun$apply$9.apply(UnfoldResourceSourceSpec.scala:182)
	at akka.stream.testkit.Utils$.assertAllStagesStopped(Utils.scala:29)
	at akka.stream.scaladsl.UnfoldResourceSourceSpec$$anonfun$1$$anonfun$apply$mcV$sp$9.apply(UnfoldResourceSourceSpec.scala:182)
	at akka.stream.scaladsl.UnfoldResourceSourceSpec$$anonfun$1$$anonfun$apply$mcV$sp$9.apply(UnfoldResourceSourceSpec.scala:182)
	at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
```

--- a/akka-stream-tests/src/test/scala/akka/stream/scaladsl/UnfoldResourceSourceSpec.scala
+++ b/akka-stream-tests/src/test/scala/akka/stream/scaladsl/UnfoldResourceSourceSpec.scala
@@ -180,20 +180,18 @@ class UnfoldResourceSourceSpec extends StreamSpec(UnboundedMailboxConfig) {
     }
 
     "fail when close throws exception" in assertAllStagesStopped {
-      val p = Source.unfoldResource[String, BufferedReader](
-        () ⇒ newBufferedReader(),
-        reader ⇒ Option(reader.readLine()),
-        reader ⇒ throw TE(""))
-        .runWith(Sink.asPublisher(false))
-      val c = TestSubscriber.manualProbe[String]()
-      p.subscribe(c)
-
-      val sub = c.expectSubscription()
-      sub.request(61)
+      val out = TestSubscriber.probe[String]()
 
       EventFilter[TE](occurrences = 1).intercept {
-        c.expectNextN(60)
-        c.expectError(TE(""))
+        Source.unfoldResource[String, Iterator[String]](
+          () ⇒ Iterator("a"),
+          it ⇒ if (it.hasNext) Some(it.next()) else None,
+          _ ⇒ throw TE(""))
+          .runWith(Sink.fromSubscriber(out))
+
+        out.request(61)
+        out.expectNext("a")
+        out.expectError(TE(""))
       }
     }
 

